client/plots/matrix.groups.js

```diff
@@ -275,14 +276,10 @@ export function classifyValues(anno, tw, grp, s, sample) {
 				} else {
 					// groupsetting in use
 					// values are group assignments
-					// only count assignments to group with highest priority
-					// in groupset
-
+					// only count assignments to group with highest
+					// priority in groupset
 					const groupset =
-						'predefined_groupset_idx' in tw.q.groupsetting
-							? tw.term.groupsetting.lst[tw.q.groupsetting.predefined_groupset_idx]
-							: tw.q.groupsetting.customset
-
+						tw.q.type == 'predefined-groupset' ? tw.term.groupsetting.lst[tw.q.predefined_groupset_idx] : tw.q.customset
 					const group = groupset.groups[0]
```

Comment:
please check if `groupset` can be undefined.
 test with [termdbtest](http://localhost:3000/?noheader=1&mass=%7B%22dslabel%22:%22TermdbTest%22,%22genome%22:%22hg38-test%22,%22nav%22:%7B%22header_mode%22:%22hidden%22%7D,%22plots%22:%5B%7B%22chartType%22:%22matrix%22,%22termgroups%22:%5B%7B%22lst%22:%5B%7B%22term%22:%7B%22gene%22:%22TP53%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22AKT1%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22id%22:%22efs%22,%22q%22:%7B%22mode%22:%22continuous%22%7D%7D,%7B%22id%22:%22agedx%22,%22q%22:%7B%22mode%22:%22continuous%22%7D%7D,%7B%22id%22:%22diaggrp%22%7D,%7B%22term%22:%7B%22gene%22:%22TP53%22,%22type%22:%22geneExpression%22%7D%7D%5D%7D%5D,%22divideBy%22:%7B%22id%22:%22sex%22%7D%7D%5D%7D), Genes > edit current group > select a msigdb set and submit

Reply:
> please check if `groupset` can be undefined. test with [termdbtest](http://localhost:3000/?noheader=1&mass=%7B%22dslabel%22:%22TermdbTest%22,%22genome%22:%22hg38-test%22,%22nav%22:%7B%22header_mode%22:%22hidden%22%7D,%22plots%22:%5B%7B%22chartType%22:%22matrix%22,%22termgroups%22:%5B%7B%22lst%22:%5B%7B%22term%22:%7B%22gene%22:%22TP53%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22AKT1%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22id%22:%22efs%22,%22q%22:%7B%22mode%22:%22continuous%22%7D%7D,%7B%22id%22:%22agedx%22,%22q%22:%7B%22mode%22:%22continuous%22%7D%7D,%7B%22id%22:%22diaggrp%22%7D,%7B%22term%22:%7B%22gene%22:%22TP53%22,%22type%22:%22geneExpression%22%7D%7D%5D%7D%5D,%22divideBy%22:%7B%22id%22:%22sex%22%7D%7D%5D%7D), Genes > edit current group > select a msigdb set and submit

Just pushed a fix